src/auth/OAuth.ts

```diff
@@ -45,17 +45,211 @@ interface ClientCredentials {
   client_assertion_type?: 'urn:ietf:params:oauth:client-assertion-type:jwt-bearer';
 }
 
+export interface AuthorizationCodeGrantRequest extends ClientCredentials {
+  /**
+   * The Authorization Code received from the initial `/authorize` call.
+   */
+  code: string;
+  /**
+   *	This is required only if it was set at the `/authorize` endpoint. The values must match.
+   */
+  redirect_uri?: string;
+}
+
+export interface AuthorizationCodeGrantWithPKCERequest extends AuthorizationCodeGrantRequest {
+  /**
+   * Cryptographically random key that was used to generate the code_challenge passed to `/authorize`.
+   */
+  code_verifier: string;
+}
+
 export interface ClientCredentialsGrantRequest extends ClientCredentials {
   /**
    * The unique identifier of the target API you want to access.
    */
   audience: string;
 }
 
+export interface PasswordGrantRequest extends ClientCredentials {
+  /**
+   * The unique identifier of the target API you want to access.
+   */
+  audience?: string;
+  /**
+   * Resource Owner's identifier, such as a username or email address.
+   */
+  username: string;
+  /**
+   * Resource Owner's secret.
+   */
+  password: string;
+  /**
+   * String value of the different scopes the application is asking for. Multiple scopes are separated with whitespace.
+   */
+  scope?: string;
+  /**
+   * String value of the realm the user belongs. Set this if you want to add realm support at this grant.
+   * For more information on what realms are refer to https://auth0.com/docs/get-started/authentication-and-authorization-flow/resource-owner-password-flow#realm-support.
+   */
+  realm?: string;
+}
+
+export interface DeviceCodeGrantRequest {
+  /**
+   * Specify this to override the parent class's `clientId`
+   */
+  client_id?: string;
+
+  /**
+   * The device code previously returned from the `/oauth/device/code` endpoint.
+   */
+  device_code: string;
+}
+
+export interface RefreshTokenGrantRequest extends ClientCredentials {
+  /**
+   * Specify this to override the parent class's `clientId`
+   */
+  client_id?: string;
```

Comment:
I would think we can drop this, as this is already defined in `ClientCredentials`

Reply:
7297be2